migrations/capcons/capabilities.go

```diff
@@ -60,6 +63,34 @@ func (c *AccountCapabilities) Record(
 	)
 }
 
+func (c *AccountCapabilities) ForEach(
+	f func(AccountCapability) bool,
+) {
+	slices.SortFunc(
+		c.Capabilities,
+		func(a, b AccountCapability) int {
+			pathA := a.TargetPath
+			pathB := b.TargetPath
+
+			if pathA.Domain == pathB.Domain {
+				return strings.Compare(pathA.Identifier, pathB.Identifier)
+			}
+
+			if pathA.Domain < pathB.Domain {
+				return -1
+			}
+
+			return +1
```

Comment:
Can maybe use Go 1.22's new `cmp.Or` and `cmp.Compare`, see e.g. https://pkg.go.dev/cmp#example-Or-Sort

Reply:
good idea! added in https://github.com/onflow/cadence/pull/3542/commits/f712fdea18aa174f9073a26156c7d101f891faba